operator.go

```diff
@@ -64,55 +78,85 @@ var (
 				return filter.Where(tx, column+" IS NULL")
 			},
 			RequiredArguments: 0,
 		},
 		"$istrue": {
-			Function: func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-				if dataType != schema.Bool {
-					return tx
+			Function: func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
+				if dataType != DataTypeBool {
+					return tx.Where("FALSE")
 				}
 				return filter.Where(tx, column+" IS TRUE")
 			},
 			RequiredArguments: 0,
 		},
 		"$isfalse": {
-			Function: func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-				if dataType != schema.Bool {
-					return tx
+			Function: func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
+				if dataType != DataTypeBool {
+					return tx.Where("FALSE")
 				}
 				return filter.Where(tx, column+" IS FALSE")
 			},
 			RequiredArguments: 0,
 		},
 		"$notnull": {
-			Function: func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
+			Function: func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
 				return filter.Where(tx, column+" IS NOT NULL")
 			},
 			RequiredArguments: 0,
 		},
 		"$between": {
-			Function: func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-				query := column + " BETWEEN ? AND ?"
-				return filter.Where(tx, query, filter.Args[0], filter.Args[1])
+			Function: func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
+				if dataType.IsArray() {
+					return tx.Where("FALSE")
+				}
+				args, ok := ConvertArgsToSafeType(filter.Args[:2], dataType)
+				if !ok {
+					return tx.Where("FALSE")
+				}
+				query := castEnumAsText(column, dataType) + " BETWEEN ? AND ?"
+				return filter.Where(tx, query, args...)
 			},
 			RequiredArguments: 2,
 		},
 	}
 )
 
-func basicComparison(op string) func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-	return func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-		query := fmt.Sprintf("%s %s ?", column, op)
-		return filter.Where(tx, query, filter.Args[0])
+func castEnumAsText(column string, dataType DataType) string {
+	if dataType == DataTypeEnum || dataType == DataTypeEnumArray {
+		return fmt.Sprintf("CAST(%s AS TEXT)", column)
 	}
+	return column
 }
 
-func multiComparison(op string) func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-	return func(tx *gorm.DB, filter *Filter, column string, dataType schema.DataType) *gorm.DB {
-		query := fmt.Sprintf("%s %s ?", column, op)
-		return filter.Where(tx, query, filter.Args)
+func basicComparison(op string) func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
+	return func(tx *gorm.DB, filter *Filter, column string, dataType DataType) *gorm.DB {
+		if dataType.IsArray() {
+			return tx.Where("FALSE")
```

Comment:
```suggestion
			return filter.Where(tx, "FALSE")
```
Fix this everywhere (including the README). Otherwise it will generate `AND` clauses even with `or` filters.